src/data/real/pi.lean

```diff
@@ -142,4 +142,121 @@ lemma pi_lt_3141593 : pi < 3.141593 := by pi_upper_bound [
   27720/19601, 56935/30813, 49359/25163, 258754/130003, 113599/56868, 1101994/551163,
   8671537/4336095, 3877807/1938940, 52483813/26242030, 56946167/28473117, 23798415/11899211]
 
+
+noncomputable theory
+open real filter
+open_locale classical big_operators topological_space
+local notation `|`x`|` := abs x
+local notation `π` := real.pi
+
+theorem mvt (f f':ℝ→ℝ) {a b c:ℝ} (h : a ≤ b) (hderiv : ∀ (x ∈ set.Icc a b), has_deriv_at f (f' x) x)
+  (hbound : ∀ (x ∈ set.Icc a b), |f' x| ≤ c) : |f b - f a| ≤ c * (b - a) :=
+begin
+  rw ← norm_eq_abs (f b - f a),
+  apply norm_image_sub_le_of_norm_deriv_le_segment',
+  { intros x hx,
+    exact (hderiv x hx).has_deriv_within_at },
+  { intros x hx,
+    rw norm_eq_abs (f' x),
+    exact hbound x (set.Ico_subset_Icc_self hx) },
+  { exact set.right_mem_Icc.mpr h },
+end
+
+
+/-! ### Leibniz's Series for Pi
+`∑ (-1)^i / (2*i+1)` from `i=1` to `n` tends to `π/4` as `n` tends to `+∞` -/
```

Comment:
This should be a docstring, not (not only?) a section title.

Please add comments explaining how the proof works.